shipgen/MuDISGenerator.cxx

```diff
@@ -347,7 +377,7 @@ Bool_t MuDISGenerator::ReadEvent(FairPrimaryGenerator* cpg)
         LOG(DEBUG) << "muon DIS Generator out part pos " << xmu << " " << ymu << "" << zmu;
         LOG(DEBUG) << "muon DIS Generator out part w " << w;
 
-        if (index == 0) {
+        if (int(mu[0][0]) == int((*Part)[0])) {
```

Comment:
Are you sure these elements always exist? That could potentially cause a segmentation violation.